protocol/xmrmaker/instance.go

```diff
@@ -198,6 +188,63 @@ func (inst *Instance) createOngoingSwap(s *swap.Info) error {
 	return nil
 }
 
+// completeSwap is called in the case where we find an ongoing swap in the db on startup,
+// and the swap already has the counterpary's swap secret stored.
+// In this case, we simply re-claim the XMR we locked, as we have both secrets required.
+// It's unlikely for this case to ever be hit, unless the daemon was shut down in-between
+// us finding the counterparty's secret and claiming the XMR.
+//
+// Note: this will use the current value of `transferBack `(verses whatever value was set when
+// the swap was started). It will also only only recover to the primary wallet address,
+// not whatever address was used when the swap was started.
+func (inst *Instance) completeSwap(s *swap.Info, skA *mcrypto.PrivateSpendKey) error {
+	// calc counterparty's swap private view key
+	// TODO: if the counterparty sends a non-standard public view key, does this work?
+	vkA, err := skA.View()
+	if err != nil {
+		return err
+	}
+
+	// fetch our swap private spend key
+	skB, err := inst.backend.RecoveryDB().GetSwapPrivateKey(s.ID)
+	if err != nil {
+		return err
+	}
+
+	// fetch our swap private view key
+	vkB, err := skB.View()
+	if err != nil {
+		return err
+	}
+
+	kpAB := pcommon.GetClaimKeypair(
+		skA, skB,
+		vkA, vkB,
+	)
+
+	_, err = pcommon.ClaimMonero(
+		inst.backend.Ctx(),
+		inst.backend.Env(),
+		s.ID,
+		inst.backend.XMRClient(),
+		s.MoneroStartHeight,
+		kpAB,
+		inst.backend.XMRClient().PrimaryAddress(),
+		true, // always sweep back to our primary address
+	)
+	if err != nil {
+		return err
+	}
+
+	s.Status = types.CompletedSuccess
```

Comment:
We are in the XMR Maker's code.  Should that be `CompletedRefund` above?

Reply:
yes, updated